server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -1054,10 +1071,8 @@ public Metadata.Builder writeTo(Metadata metadata,
             }
             indexUUIDs.add(indexMetadata.getIndexUUID());
 
-            long allowedTotalColumns = TOTAL_COLUMNS_LIMIT.get(indexMetadata.getSettings());
-            if (allColumns.size() > allowedTotalColumns) {
-                throw new IllegalArgumentException("Limit of total columns [" + allowedTotalColumns + "] in table [" + ident + "] exceeded");
-            }
+            final Settings indexSettings = indexMetadata.getSettings();
+            checkColumnLimits(ident, indexSettings, allColumns);
```

Comment:
If it's checked here, is it also necessary to check it in the `TransportCreateTable`?

Reply:
Thank you @matriv . As Jordi pointed out, this will prevent i.e. `alter` on a table created before `6.0` containing objects with depth > 100. So I will remove this.

BTW, the check in `TransportCreateTable` https://github.com/crate/crate/pull/17845/files#diff-ce9aa6544f3ae05c900574e9268de41b21d3fd1512348dd60a7b85e94491087bR153-R162 is being added, do you think it is not enough?